src/main/java/com/obal/dominos/App.java

```diff
@@ -1,19 +1,102 @@
 package com.obal.dominos;
 
-import java.util.Scanner;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedList;
 
-/**
- * Hello world!
- *
- */
+class Domino {
+    byte[] values;
+
+    public Domino(byte[] vl) {
+        values = vl;
+    }
+
+    void printValue() {
+        System.out.println(Arrays.toString(values));
+    }
+}
+
+class Hand {
+    LinkedList<Domino> dominoes = new LinkedList<>();
+
+    void printHand() {
+        for (Domino domino : dominoes)
+            System.out.print(Arrays.toString(domino.values));
+        System.out.println();
+    }
+
+    void addDomino(Domino domino) {
+        dominoes.addFirst(domino);
+    }
+    boolean isInHand(Domino domino) {
+        return dominoes.contains(domino);
+    }
+    void dropDomino(Domino domino) {
+        if (isInHand(domino))
+            dominoes.removeFirstOccurrence(domino);
+        //TODO : ADD ERROR HANDLING
+    }
+}
+
+class Snake {
+    // TODO : Find a better name for dominoes on the board haha
+    LinkedList<Domino> dominoes = new LinkedList<>();
+
+    void printSnake() {
+        for (Domino domino : dominoes)
+            System.out.print(Arrays.toString(domino.values));
+        System.out.println();
+    }
+
+    byte getRightValue() {
+        return dominoes.getLast().values[1];
```

Comment:
we actually don't know if the last item of values will be the free end of the snake, the player might lay down the domino in the other way. I suggest keeping track of the snakes ends, updating it as new dominos get laid.